.github/workflows/publish_docs.yml

```diff
@@ -0,0 +1,18 @@
+name: 'publish docs'
+
+# Publish when a git tag is pushed to main
+on:
+  push:
+    branches:
+      - main
+    tags:
+      - '*'
+
+jobs:
+  publish_sphinx_docs:
```

Comment:
Should this be merged with `check_doc.yml` and linked with a `needs`? I.e....
```
check_sphinx_docs → publish_sphinx_docs
```

Reply:
I ended up getting rid of `check_docs.yml` for now (see reasons in my extensive comment)